power_grid_model_c/power_grid_model/include/power_grid_model/component/transformer_tap_regulator.hpp

```diff
@@ -35,10 +35,10 @@ class TransformerTapRegulator : public Regulator {
     UpdateChange update(TransformerTapRegulatorUpdate const& update_data) {
         assert(update_data.id == id());
         set_status(update_data.status);
-        u_set_ = update_data.u_set;
-        u_band_ = update_data.u_band;
-        line_drop_compensation_r_ = update_data.line_drop_compensation_r;
-        line_drop_compensation_x_ = update_data.line_drop_compensation_x;
+        update_real_value<symmetric_t>(update_data.u_set, u_set_, 1.0);
```

Comment:
Why was this needed? if update_data.u_set is nan, then it would have been replaced by nan anyway

Reply:
yes but if the original value is not `nan`, then the original code will make it `nan` and invalidate the input data. that is not according to our current contract with the users